src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECCng.ImportExport.cs

```diff
@@ -0,0 +1,401 @@
+                {
+                    BCRYPT_ECCFULLKEY_BLOB* pBcryptBlob = (BCRYPT_ECCFULLKEY_BLOB*)pEcBlob;
+
+                    var primeCurve = new ECCurve();
+                    primeCurve.CurveType = (ECCurve.ECCurveType)pBcryptBlob->CurveType;
+                    primeCurve.Hash = GetHashAlgorithmName(pBcryptBlob->CurveGenerationAlgId);
+
+                    int offset = sizeof(BCRYPT_ECCFULLKEY_BLOB);
+
+                    primeCurve.Prime = Interop.BCrypt.Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+                    primeCurve.A = Interop.BCrypt.Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+                    primeCurve.B = Interop.BCrypt.Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+                    primeCurve.G = new ECPoint()
+                    {
+                        X = Interop.BCrypt.Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength),
+                        Y = Interop.BCrypt.Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength),
+                    };
+                    primeCurve.Order = Interop.BCrypt.Consume(ecBlob, ref offset, pBcryptBlob->cbSubgroupOrder);
+                    primeCurve.Cofactor = Interop.BCrypt.Consume(ecBlob, ref offset, pBcryptBlob->cbCofactor);
+
+                    // Optional parameters
+                    primeCurve.Seed = pBcryptBlob->cbSeed == 0 ? null : Interop.BCrypt.Consume(ecBlob, ref offset, pBcryptBlob->cbSeed);
+
+                    ecParams.Q = new ECPoint
+                    {
+                        X = Interop.BCrypt.Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength),
+                        Y = Interop.BCrypt.Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength)
+                    };
+
+                    if (includePrivateParameters)
+                    {
+                        ecParams.D = Interop.BCrypt.Consume(ecBlob, ref offset, pBcryptBlob->cbSubgroupOrder);
+                    }
+
+                    ecParams.Curve = primeCurve;
+                    return ecParams;
+                }
+            }
+        }
+
+        /// <summary>
+        ///     This function checks the magic value in the key blob header
+        /// </summary>
+        /// <param name="includePrivateParameters">Private blob if true else public key blob</param>
+        private static void CheckMagicValueOfKey(KeyBlobMagicNumber magic, bool includePrivateParameters)
+        {
+            if (includePrivateParameters)
+            {
+                if (!IsMagicValueOfKeyPrivate(magic))
+                {
+                    throw new CryptographicException(SR.Cryptography_NotValidPrivateKey);
+                }
+            }
+            else
+            {
+                if (!IsMagicValueOfKeyPublic(magic))
+                {
+                    // Private key magic is permissible too since the public key can be derived from the private key blob.
+                    if (!IsMagicValueOfKeyPrivate(magic))
```

Comment:
If all the callers (this may be the only one) are going to have this logic, maybe the end of IsMagicValueOfKeyPublic should be `return IsMagicValueOfKeyPrivate(magic);` instead of `return false;`.  (I can see arguments for separation, and for chaining; so I'm just mentioning it as a "consider")